pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -551,33 +553,41 @@ func (c *Cluster) initMonIPs(mons []*monConfig) error {
 	return nil
 }
 
+// Delete mon canary deployments (and associated PVCs) using deployment labels
+// to select this kind of temporary deployments
+func (c *Cluster) removeCanaryDeployments() {
+	canaryDeployments, err := k8sutil.GetDeployments(c.context.Clientset, c.Namespace, "app=rook-ceph-mon,mon_canary=true")
+	if err != nil {
+		logger.Warningf("failed to get the list of monitor canary deployments. %v", err)
+		return
+	}
+
+	for _, canary := range canaryDeployments.Items {
+		logger.Infof("cleaning up canary monitor deployment %q and canary pvc %q.", canary.Name, canary.Labels["pvc_name"])
+		if err := k8sutil.DeleteDeployment(c.context.Clientset, c.Namespace, canary.Name); err != nil {
+			logger.Infof("failed to delete canary monitor deployment %q. %v", canary.Name, err)
+		}
+
+		if canary.Labels["pvc_name"] != "" {
+			var gracePeriod int64 // delete immediately
+			propagation := metav1.DeletePropagationForeground
+			options := &metav1.DeleteOptions{GracePeriodSeconds: &gracePeriod, PropagationPolicy: &propagation}
+			err := c.context.Clientset.CoreV1().PersistentVolumeClaims(c.Namespace).Delete(canary.Labels["pvc_name"], options)
+			if err != nil {
+				logger.Infof("failed to delete canary monitor %q pvc %q. %v", canary.Name, canary.Labels["pvc_name"], err)
```

Comment:
```suggestion
				logger.Warningf("failed to delete canary monitor %q pvc %q. %v", canary.Name, canary.Labels["pvc_name"], err)
```

Reply:
Yep!!! indeed more coherent ... Thank you !!!